src/types/currency.ts

```diff
@@ -1,93 +1,133 @@
-import { makeClass } from "../utils/make-class";
-const _ = require("lodash");
-const { slice } = require("../utils/bytes-utils");
-const { Hash160 } = require("./hash-160");
+import { Hash160 } from "./hash-160";
 const ISO_REGEX = /^[A-Z0-9]{3}$/;
 const HEX_REGEX = /^[A-F0-9]{40}$/;
 
-function isoToBytes(iso) {
-  const bytes = new Uint8Array(20);
+/**
+ * Convert an iso code to a currency bytes representation
+ */
+function isoToBytes(iso: string): Buffer {
+  const bytes = Buffer.alloc(20);
   if (iso !== "XRP") {
     const isoBytes = iso.split("").map((c) => c.charCodeAt(0));
     bytes.set(isoBytes, 12);
   }
   return bytes;
 }
 
-function isISOCode(val) {
-  return val.length === 3; // ISO_REGEX.test(val);
+/**
+ * Tests if iso is a valid iso code
+ */
+function isISOCode(iso: string): boolean {
+  return ISO_REGEX.test(iso);
 }
 
-function isHex(val) {
-  return HEX_REGEX.test(val);
+/**
+ * Tests if hex is a valid hex-string
+ */
+function isHex(hex: string): boolean {
+  return HEX_REGEX.test(hex);
 }
 
-function isStringRepr(val) {
-  return _.isString(val) && (isISOCode(val) || isHex(val));
+/**
+ * Tests if a string is a valid representation of a currency
+ */
+function isStringRepresentation(str: string): boolean {
```

Comment:
optional drive by: Generally prefer full named variables for readability. I would favor using `input: string` here since `input` has more semantic meaning than `str`. 

Same comments could apply to `val` below too.